### ISSUE TITLE
data-bind: checkbox not working when used in foreach binding

### ISSUE THREAD

**Issue by fieder** [faulknercs/Knockstrap#34]
Hi,
I'm using toggle buttons to enable search by 'tags'. A tag is simply a string. In my case these represent music genres i.e. 'Rock', 'Jazz', 'Blues'. I want to provide a list of buttons (topCategoriesList) that users can switch on/off by pressing it to include the tag in the search. 

My markup looks like

    <div id="topCategoryFilter" class="btn-group form-group" data-toggle="buttons" 
    data-bind="checkbox:     chosenCategories, foreach: { data: topCategoriesList, as: 'topCategory' }">

Unfortunately chosenCategories (model) never get the values

This is the toggle button itself.
    
    <label class="btn btn-primary" > 
         <input type="checkbox" value="$data" />
        <span data-bind="text: topCategory" onclick="filterScreen()"></span>
    </label>

This is the initialisation of my model

    var chosenCategories = ko.observableArray([]);

Where filterScreen is the method that checks for the observable 'chosenCategories' but nothing gets captured. I'd expect the list of strings (buttons pressed on). Any idea?

**Comment by faulknercs** [comment]
Hello,

You should use `data-bind="value: $data"` instead of `value=$data` in order to make checkboxes work.
Please, see https://jsfiddle.net/faulknercs/yrcgd8s5/ for working example.

**Comment by fieder** [comment]
Hi there, thank you very much for sharing your working example.

After trying many things I realised your working example (knockout 3.0.0) was not using the same libraries I'm using in my local project (knockout 3.3.0).

https://jsfiddle.net/fieder/r2syym59/1/  is exactly the same example you sent me but with knockout 3.3.0 - you will notice it does not work.

Having said that, I'm happy to downgrade to knockout 3.0.0 in order to make this work. I've done it already so I'm running locally knockout 3.0.0. I'm also running bootstrap 3.2.0 like your working example (I was running 3.3.5 which also works but just to make sure I've downgraded bootstrap too to have same env).

Unfortunately my code does not work yet. This is it in case you spot something I couldn't so far :(

     <div class="container">
            <div class="row">
                <div class="hf-label col-sm-10 col-sm-push-1 col-xs-12">
                    <div data-bind="checkbox: eventListPage().chosenCategories" class="topCategoryFilter">
                        <span data-toggle="buttons"
                              data-bind="foreach: { data: eventListPage().topCategoriesList}">
                            <label class="btn btn-transparent">
                                <input type="checkbox" data-bind="value: $data" />
                                <span data-bind="text: $data"></span>
                            </label>
                        </span>
                        <label class="btn btn-transparent">
                            <input type="checkbox" data-bind='checked: eventListPage().showAllCategories, click'
                                   onclick="havefun.hideTopList(this)"/>
                            <span>MORE</span>
                        </label>
                    </div>
                    <div data-bind='visible: eventListPage().showAllCategories'>
                        <span class="allCategoryFilter" data-toggle="buttons"
                              data-bind="checkbox: eventListPage().chosenCategories, foreach: { data: eventListPage().categoriesList }">
                            <label class="btn btn-transparent">
                                <input type="checkbox" data-bind="value: $data" />
                                <span data-bind="text: $data"></span>
                            </label>
                        </span>
                        <label class="btn btn-transparent">
                            <input type="checkbox" data-bind='checked: eventListPage().showAllCategories'
                                   onclick="havefun.hideTopList(this)"/>
                            <span>LESS</span>
                        </label>
                    </div>
                </div>
            </div>
        </div>

**Comment by fieder** [comment]
I thought this could be useful to describe the behaviour I see
https://www.youtube.com/watch?v=DIxECRkUQHA

Please let me know what you think. I've spent a lot of time on it and I'm running out of options...

Cheers!

**Comment by faulknercs** [comment]
Hello,

**Comment by faulknercs** [comment]
Hello,
At first I checked your jsfiddle with Knockout 3.3.0. It is not working because Knockout is added after Knockstarp. Here is correct variant https://jsfiddle.net/faulknercs/r2syym59/2/

Then I checked html code. `checkbox` binding should be used with bootstrap buttons, it means, that `data-toggle="buttons"` attribute should be at same element as `data-bind="checkbox: chosenCategories"`. It should throw exception, if there is no `data-toggle="buttons"`, I don't know, why there is no exception in console in your video :)

I have created new jsfiddle example based on your html code: http://jsfiddle.net/faulknercs/9caassvs/
It is unstyled, but I think, it is not important.
Hope, it will help you.

Best regards,
Artem
